uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -294,16 +289,16 @@ define([
 
         // Make a copy of the list
         var listToReturn = vm.leaveRequestDayTypes.slice(0);
-        date = convertDateFormatToServer(date);
 
+        date = convertDateFormatToServer(date);
         PublicHoliday.isPublicHoliday(date)
           .then(function (result) {
             if (result) {
               listToReturn = listToReturn.filter(function (publicHoliday) {
                 return publicHoliday.name === 'public_holiday';
               });
             } else {
-              var inCalendarList = getDayTypesFromDate(date);
+              var inCalendarList = getDayTypesFromDate(date, listToReturn);
```

Comment:
its a local variable relevant in the block in which its declared, so its optimal